### PR TITLE
improve new user registration

### PR DESCRIPTION
**Pull request by vairafiq**
Issue https://github.com/sovware/directorist/issues/800

### CODE REVIEW

**Review by obiPlabon — COMMENTED** [review]
We cannot simply add this check without informing the users beforehand. As the `user_can_register` is disabled by default on all the WordPress installations. So if we push this without notifying the users then there's a chance that users will not accept this change easily.

**Comment by vairafiq** [comment]
Screen Shot 2022-04-10 at 10 16 19 AM

Great! Then should we push this warning to Directorist >> System Status?

**Comment by vairafiq** [comment]
Screen Shot 2022-04-11 at 10 15 10 AM

New option introduced

File: includes/model/Account.php [hunk 1]
```diff
@@ -35,14 +35,44 @@ public function render_shortcode_login( $atts = [] ) {
...
 		ob_start();
 		if ( ! empty( $atts['shortcode'] ) ) { Helper::add_shortcode_comment( $atts['shortcode'] ); }
-		echo Helper::get_template_contents( 'account/login' );
+		echo Helper::get_template_contents( 'account/login', $args );
 
 		return ob_get_clean();
 	}
 
 	public function render_shortcode_registration( $atts ) {
+
```
Comment (obiPlabon) [inline includes/model/Account.php:75]: if ( ! get_directorist_option( 'new_user_registration', true ) ) {
    return;
}

**Review by obiPlabon — APPROVED** [review]

**Pull request merged**